File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        #using hashmap but space is O(n)
        d={}
        if len(nums)==1:
            return nums[0]
        for i in nums:
            if i in d and d[i]==len(nums)//2:                
                return i
            if i in d:                
                d[i]+=1
            else:                
                d[i]=1

File: 169-majority-element/169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) 
    {
     
        //
        
        int ele=0,count=0;
        for(int a:nums)
        {
            if (count==0)
                ele=a;
            if(ele==a)
                count++;
            else
                count--;
        }
        return ele;
    }
}